kafka/src/main/scala/filodb/kafka/package.scala

```diff
@@ -33,7 +33,11 @@ package object kafka {
   implicit final class MapOps(immutable: Map[String, AnyRef]) {
     def asProps: JProperties = {
       val props = new JProperties()
-      props.putAll(immutable.filter(_._2 != null).asJava)
+      for ((k,v) <- immutable) {
```

Comment:
Or:

    immutable.foreach { case (k, v) => if (k != null && v != null) props.put(k, v) }

Reply:
The if condition can also be included in the for comprehension.

Reply:
Which format is more efficient? In general, I try to avoid passing functions into iterators, since it creates additional objects and the introduction of a virtual call in a tight loop can slow things down as well.